### PR TITLE
Feature/7953 create content validation script

### PR DESCRIPTION
**Pull request by timwright12**
Set up Jenkins to run a11y checks on a schedule. Needs to be set up in Jenkins itself as well.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,22 +1,39 @@
 version: 2.1
```
Comment (timwright12) [inline .circleci/config.yml:1]: This whole file can be ignored until we're ready to wire up circle

File: Jenkinsfile [hunk 2]
```diff
@@ -21,74 +21,73 @@ node('vetsgov-general-purpose') {
...
+      // commonStages.slackNotify()
+      throw error
+    } finally {
+      dir("content-build") {
+        step([$class: 'JUnitResultArchiver', testResults: 'test-results.xml'])
       }
     }
   }
 
+  // // Run E2E and accessibility tests
```
Comment (timwright12) [inline Jenkinsfile:65]: Just commenting out for now, I moved it into another file as well to run on a schedule

File: Jenkinsfile.integration [hunk 3]
```diff
@@ -1,31 +1,39 @@
...
+          'nightwatch-accessibility': {
+            sh "export IMAGE_TAG=${commonStages.IMAGE_TAG} && docker-compose -p accessibility up -d && docker-compose -p accessibility run --rm --entrypoint=npm -e BABEL_ENV=test -e BUILDTYPE=vagovprod content-build --no-color run nightwatch:docker -- --env=accessibility"
           },
-          // jest: {
-          //   sh "docker-compose -p e2e run --rm --entrypoint npm -e BABEL_ENV=test -e BUILDTYPE=vagovprod vets-website --no-color run test:integration:jest"
-          // }
         )
       } catch (error) {
-        // commonStages.slackIntegrationNotify()
+        // commonStages.slackNotify()
```
Comment (timwright12) [inline Jenkinsfile.integration:32]: Not notifying slack at this time

File: jenkins/common.groovy [hunk 4]
```diff
@@ -17,9 +17,9 @@ DRUPAL_CREDENTIALS = [
 ]
 
 ALL_VAGOV_BUILDTYPES = [
```
Comment (timwright12) [inline jenkins/common.groovy:19]: turning all these back on since they're just builds

File: jenkins/common.groovy [hunk 5]
```diff
@@ -190,7 +191,7 @@ def build(String ref, dockerContainer, String assetSource, String envName, Boole
       sh "cd /application && jenkins/build.sh --envName ${envName} --assetSource ${assetSource} --drupalAddress ${drupalAddress} ${drupalMode} --buildLog ${buildLogPath} --verbose"
 
       if (envName == 'vagovprod') {
-	checkForBrokenLinks(buildLogPath, envName, contentOnlyBuild)
+	       checkForBrokenLinks(buildLogPath, envName, contentOnlyBuild)
```
Comment (timwright12) [inline jenkins/common.groovy:194]: only check links on prod to save some time

File: Jenkinsfile.integration (hunk 3 above)
Comment (billfienberg) [reply Jenkinsfile.integration:32]: Do you remember how/where Slack notification is configured? I just want to make sure we're not alarming/confusing other teams.

File: jenkins/common.groovy [hunk 6]
```diff
@@ -45,7 +45,8 @@ def isReviewable() {
 def isDeployable() {
   return (IS_DEV_BRANCH ||
           IS_STAGING_BRANCH) &&
-    !env.CHANGE_TARGET
+    !env.CHANGE_TARGET &&
+    !currentBuild.nextBuild // if there's a later build on this job (branch), don't deploy
```
Comment (billfienberg) [inline jenkins/common.groovy:49]: I think the `!currentBuild.nextBuild` condition was just removed from `vets-website` in https://github.com/department-of-veterans-affairs/vets-website/pull/15487. Would we want to also remove that condition here?

File: Jenkinsfile.integration (hunk 3 above)
Comment (timwright12) [reply Jenkinsfile.integration]: This one is set up in the groovy file as far as I can tell

File: jenkins/common.groovy (hunk 6 above)
Comment (timwright12) [reply jenkins/common.groovy:49]: yeah I'm not sure why it's picking this up, nice catch

File: jenkins/common.groovy [hunk 7]
```diff
@@ -216,7 +216,7 @@ def buildAll(String ref, dockerContainer, Boolean contentOnlyBuild) {
         def envName = VAGOV_BUILDTYPES.get(i)
         builds[envName] = {
           try {
-            build(ref, dockerContainer, assetSource, envName, false, contentOnlyBuild)
+            build(ref, dockerContainer, assetSource, envName, false, false, contentOnlyBuild)
```
Comment (billfienberg) [inline jenkins/common.groovy:219]: I'm probably missing something, but I just want to make sure. 

Doesn't the [`build` function](https://github.com/department-of-veterans-affairs/content-build/pull/42/files#diff-2cc6a42e83abdf9666365745cd63a9a93da477f58972350809e0b2acd32dfaceR178) only take 6 positional arguments? 

```
def build(String ref, dockerContainer, String assetSource, String envName, Boolean useCache, Boolean contentOnlyBuild)
```

If I'm reading it correctly, I think your change introduces a seventh positional argument.

File: src/platform/testing/e2e/mockapi.js [hunk 8]
```diff
@@ -0,0 +1,146 @@
+'use strict'; // eslint-disable-line strict
```
Comment (billfienberg) [inline src/platform/testing/e2e/mockapi.js:1]: This file was copied from https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/testing/e2e/mockapi.js, correct?

File: jenkins/common.groovy (hunk 6 above)
Comment (timwright12) [reply jenkins/common.groovy]: should be good now

File: jenkins/common.groovy (hunk 7 above)
Comment (timwright12) [reply jenkins/common.groovy]: yeah, I think this was a leftover from the validation argument I added, removed it

File: src/platform/testing/e2e/mockapi.js (hunk 8 above)
Comment (timwright12) [reply src/platform/testing/e2e/mockapi.js:1]: yes

File: script/run-mockapi.sh [hunk 9]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
```
Comment (billfienberg) [inline script/run-mockapi.sh:1]: This file was copied from [`vets-website/blob/master/script/run-mockapi.sh`](https://github.com/department-of-veterans-affairs/vets-website/blob/master/script/run-mockapi.sh), correct?

Comment (timwright12) [reply script/run-mockapi.sh:1]: correct

**Review by billfienberg — APPROVED** [review]
Ship it

**Pull request merged**